### DISCUSSION TITLE
What is the difference between Vanna and a comprehensive system-prompt

### DISCUSSION THREAD

**Discussion by Joly0** [vanna-ai/vanna#546] [General]
Hey guys, we are currently trying to check, if vanna-ai is a valid product to test for txt2sql chatting. We have tried multiple attempts and found that vanna-ai works good, but we also wrote a program our-selves that uses a comprehensive and specific system prompt for the llm. Now i am asking myself, what is vanna doing differently and providing differently, than our approach? As far as i can see, vanna is also just creating a comprehensive system-prompt to the llm based on the ddl, documentation and q/a´s or is it doing anything more?

**Comment by zainhoda** [comment]
If you have a very simple database and everything fits into the context window, then you probably don't need Vanna! Vanna helps more when you have specific goals for accuracy rates and need lots of question-SQL pairs as part of the prompt. 

The other thing that Vanna gives you is the abstractions for sql generation, chart generation, and fitting that into various interfaces.

**Reply by Bakulesh1Codes108** [reply]
My Organisation is building on Langchain but as asked me to find merits   of using Vanna Text to SQL ? which one would be efficient data size is huge running into 100 Millions multi table and its Muti  product also Time series Data  monthly loading data 

What I realised studying chunking and indexing RAG Documents  providing  the RAG pipeline data gives more than 90 %  accuracy even over costly GraphRags for not missing out on  semantic meanings 
Also  Vanna ai uses and can read  and uses Database Schemas  – also known as ERDs or Entity-Relationship Diagrams 

 I want to  know how Vanna ai uses it that (Schema-Relationships) ?
What are merits of using it over Langchain  & LamdaIndex ?  Users on reddit argued on Based level controls and infrastructure for maintenance (They mentioned Vanna ai don't have )?

I want them to implement Vanna ai need to show them with Benchmark Index (Professional PPT) and also how Vanna fares on similar index like BIRD (BIg Bench for LaRge-scale Database Grounded Text-to-SQL Evaluation)

Please also point me towards  better architecture and how KGs Knowledge Graphs like stuffs could be added into Vanna AI .

**Reply by Bakulesh1Codes108** [reply]
Where and where modification is possible with indexing and chunking processes ?

**Reply by Bakulesh1Codes108** [reply]
I need points against lamdaindex and langchain

**Reply by Bakulesh1Codes108** [reply]
Where the function or is there any already existing function which asks Vanna AI to store Database Detailed Schema
